### PR TITLE
momentjs/date-fns to dayjs

### PR DESCRIPTION
**Pull request by akanshaaa19**
## Summary

Migrated to dayjs from  date-fns / momentjs as now it's a recommended option by MUI. 
Check https://mui.com/x/react-date-pickers/#date-library
Datejs: https://day.js.org/

### CODE REVIEW

**Review by mdshamoon — CHANGES_REQUESTED** [review]
Overall looks good.

Two things:
- We should remove date-fns also with dayjs as mentioned in the issue. There is still a date-fns adapter that is being used
- Remove moment and date-fns packages from package.json

File: src/components/UI/Form/Calendar/Calendar.tsx [hunk 1]
```diff
@@ -31,7 +31,7 @@ export const Calendar = ({
   const [open, setOpen] = useState(false);
 
   const handleDateChange = (date: Date | null | string) => {
-    if (date) {
+    if (date) {
```
Comment (kurund) [inline src/components/UI/Form/Calendar/Calendar.tsx:34]: @akanshaa19 Can you look at your editor's formatting?

File: src/containers/Chat/ChatConversations/ChatConversations.tsx [hunk 2]
```diff
@@ -34,7 +34,7 @@ export const ChatConversations = ({ contactId }: ChatConversationsProps) => {
   const offset = useQuery(SEARCH_OFFSET);
   const client = useApolloClient();
   const { t } = useTranslation();
-
+
```
Comment (kurund) [inline src/containers/Chat/ChatConversations/ChatConversations.tsx:37]: same as above.

File: src/containers/List/List.tsx [hunk 3]
```diff
@@ -493,7 +493,7 @@ export const List = ({
       );
     }
     return null;
-  }
+  }
```
Comment (kurund) [inline src/containers/List/List.tsx:496]: same as above

File: src/common/constants.ts [hunk 4]
```diff
@@ -11,7 +11,7 @@ export const FLOW_STATUS_PUBLISHED = 'published';
 export const SIMULATOR_NUMBER_START = '9876543210';
 export const GUPSHUP_ENTERPRISE_SHORTCODE = 'gupshup_enterprise';
 export const VALID_URL_REGEX =
-  'https?://(www.)?[-a-zA-Z0-9@:%._+~#=]{2,256}.[a-z]{2,4}([-a-zA-Z0-9@:%_+.~#?&\/=]*)';
+  'https?://(www.)?[-a-zA-Z0-9@:%._+~#=]{2,256}.[a-z]{2,4}([-a-zA-Z0-9@:%_+.~#?&/=]*)';
```
Comment (kurund) [inline src/common/constants.ts:14]: was this an intentional change?

File: src/common/constants.ts [hunk 5]
```diff
@@ -112,8 +112,8 @@ export const setVariables = (
   },
 });
 
-export const is24HourWindowOver = (time: any) =>
-  moment.duration(moment(new Date()).diff(moment(time))).asHours() > 24;
+export const is24HourWindowOver = (time: any) =>
```
Comment (kurund) [inline src/common/constants.ts:115]: extra space here too.

File: src/common/constants.ts (hunk 4 above)
Comment (akanshaaa19) [reply src/common/constants.ts:14]: i guess prettier plugin did that

File: src/containers/SettingList/OrganizationFlows/OrganisationFlows.tsx [hunk 6]
```diff
@@ -75,8 +76,8 @@ export const OrganisationFlows = () => {
   const getEnabledDays = (data: any) => data.filter((option: any) => option.enabled);
 
   const setOutOfOffice = (data: any) => {
-    setStartTime(data.startTime);
-    setEndTime(data.endTime);
+    setStartTime(dayjs(`${dayjs().format('YYYY-MM-DD')}T${data.startTime}`));
```
Comment (mdshamoon) [inline src/containers/SettingList/OrganizationFlows/OrganisationFlows.tsx:79]: Just test this out once if its working fine.

File: src/components/UI/Form/DateTimePicker/DateTimePicker.tsx [hunk 7]
```diff
@@ -17,7 +19,7 @@ export interface DateTimePickerProps {
 }
 
 export const DateTimePicker = ({
-  format = 'dd/MM/yyyy hh:mm a',
+  format = 'DD/MM/YYYY hh:mm a',
```
Comment (mdshamoon) [inline src/components/UI/Form/DateTimePicker/DateTimePicker.tsx:22]: We should use the constant here.

File: src/mocks/Trigger.tsx [hunk 8]
```diff
@@ -77,7 +78,7 @@ export const getTriggerQuery = (frequency: any) => ({
       trigger: {
         trigger: {
           days: [1, 2],
-          endDate: '2021-03-13',
+          endDate: dayjs('2021-03-13'),
```
Comment (mdshamoon) [inline src/mocks/Trigger.tsx:81]: This is a query mock. It represents the data that will be sent by the backend using a particular query. In this case backend will send back a date string only and not a dayjs object. Please update this.

File: src/mocks/Trigger.tsx [hunk 9]
```diff
@@ -90,7 +91,7 @@ export const getTriggerQuery = (frequency: any) => ({
           id: '1',
           isActive: true,
           isRepeating: true,
-          startAt: '2021-02-28T20:00:22Z',
+          startAt: dayjs('2021-02-28T20:00:22Z'),
```
Comment (mdshamoon) [inline src/mocks/Trigger.tsx:94]: Same here.

File: src/common/constants.ts [hunk 10]
```diff
@@ -1,17 +1,25 @@
-import moment from 'moment';
+import dayjs from 'dayjs';
 import * as Yup from 'yup';
 
 export const SIDE_DRAWER_WIDTH = 233;
 export const DATE_FORMAT = 'DD/MM/YY';
 export const FULL_DATE_FORMAT = 'DD/MM/YYYY';
+export const FULL_DATE_FORMAT2 = 'YYYY-MM-DD';
+export const FULL_DATE_FORMAT_WITH_MONTH = 'DD MMM YYYY';
```
Comment (mdshamoon) [inline src/common/constants.ts:8]:
```suggestion
export const SHORT_DATE_FORMAT = 'DD/MM/YY';
export const LONG_DATE_FORMAT = 'DD/MM/YYYY';
export const ISO_DATE_FORMAT = 'YYYY-MM-DD';
export const DATE_FORMAT_WITH_MONTH = 'DD MMM YYYY';
```

File: src/common/constants.ts [hunk 11]
```diff
@@ -1,17 +1,25 @@
...
 export const TIME_FORMAT = 'HH:mm';
-export const DATE_TIME_FORMAT = 'DD/MM/YYYY, HH:mm:ss';
+export const FULL_TIME_FORMAT = 'HH:mm:ss';
+export const FULL_TIME_FORMAT2 = 'THH:mm:ss';
+export const FULL_DATE_TIME_FORMAT = 'DD/MM/YYYY, HH:mm:ss';
+export const DATE_TIME_FORMAT = 'DD/MM/YYYY, HH:mm';
+export const DATE_TIME_FORMAT2 = 'MMMM DD, YYYY, [at] HH:mm:ss';
+export const DATE_TIME_FORMAT_WITH_AMPM = 'DD/MM/YYYY_hh:mmA';
+export const DATE_TIME_FORMAT_WITH_AMPM2 = 'DD MMM YYYY hh:mm a';
```
Comment (mdshamoon) [inline src/common/constants.ts:16]:
```suggestion
export const SHORT_TIME_FORMAT = 'HH:mm';
export const LONG_TIME_FORMAT = 'HH:mm:ss';
export const EXTENDED_TIME_FORMAT = 'THH:mm:ss';
export const STANDARD_DATE_TIME_FORMAT = 'DD/MM/YYYY, HH:mm:ss';
export const SHORT_DATE_TIME_FORMAT_ = 'DD/MM/YYYY, HH:mm';
export const EXTENDED_DATE_TIME_FORMAT = 'MMMM DD, YYYY, [at] HH:mm:ss';
export const DATE_TIME_FORMAT_WITH_AMPM_SHORT = 'DD/MM/YYYY_hh:mmA';
export const DATE_TIME_FORMAT_WITH_AMPM_LONG = 'DD MMM YYYY hh:mm a';
```

**Review by mdshamoon — CHANGES_REQUESTED** [review]

**Review by mdshamoon — APPROVED** [review]

**Pull request merged**